### DISCUSSION TITLE
Using with Laravel Sail

### DISCUSSION THREAD

**Discussion by garbinmarcelo** [spatie/browsershot#591] [Q&A]
Hi,

How can I use this pack with Laravel Sail? Any tips for installing with the Sail build?

I tried to add the following to the Dockerfile:
```dockerfile
# Install Chrome
RUN apt-get update \
    && apt-get install -y wget \
    && wget -q -O - https://dl-ssl.google.com/linux/linux_signing_key.pub | apt-key add - \
    && sh -c 'echo "deb http://dl.google.com/linux/chrome/deb/ stable main" >> /etc/apt/sources.list.d/google.list' \
    && apt-get update \
    && apt-get -f install \
    && apt-get install google-chrome-stable
```

But there is an error, I've been doing this for a while, all help is welcome.

Thanks

**Comment by rostockahoi** [comment]
At first you have to publish Sail's Dockerfiles. Then put the following into the Dockerfile
```bash
# ...
&& curl -sL https://deb.nodesource.com/setup_16.x | bash - \
&& apt-get install -y nodejs gconf-service libasound2 libatk1.0-0 libc6 libcairo2 libcups2 libdbus-1-3 libexpat1 libfontconfig1 libgbm1 libgcc1 libgconf-2-4 libgdk-pixbuf2.0-0 libglib2.0-0 libgtk-3-0 libnspr4 libpango-1.0-0 libpangocairo-1.0-0 libstdc++6 libx11-6 libx11-xcb1 libxcb1 libxcomposite1 libxcursor1 libxdamage1 libxext6 libxfixes3 libxi6 libxrandr2 libxrender1 libxss1 libxtst6 ca-certificates fonts-liberation libappindicator1 libnss3 lsb-release xdg-utils wget libgbm-dev libxshmfence-dev \
&& npm install --global --unsafe-perm puppeteer \
&& chmod -R o+rx /usr/lib/node_modules/puppeteer/.local-chromium \
# ...
```

Then rebuild the containers.

Bear in mind that there is a caveat when using Browsershot and Sail. You can read about it here: https://github.com/spatie/browsershot/discussions/516#discussioncomment-2013859

**Reply by johncarter-** [reply]
Thanks for the Dockerfile addition, when I run the `sail build --nocache` command I eventually get a `chmod: cannot access '/usr/lib/node_modules/puppeteer/.local-chromium': No such file or directory`.

I can see that `/usr/lib/node_modules/puppeteer/` does exist, but indeed that `.local-chromium` isn't there.

Do you have any clues as to why that has happened? I've tried it a couple of times.

**Reply by johncarter-** [reply]
Related to: https://github.com/spatie/browsershot/discussions/681

**Reply by fergus-ugle** [reply]
@johncarter- did you manage to solve this? Having the same issue

**Reply by johncarter-** [reply]
I'm afraid I can't remember which project this was from. I suspect I just span up a staging server on the actual host.

**Reply by michaelklopf** [reply]
Does this help? https://laracasts.com/discuss/channels/laravel/run-browsershot-in-laravel-sail-on-arm64-mac

**Reply by fergus-ugle** [reply]
@michaelklopf yes this solves the issue! I followed your solution on an Ubantu machine and worked perfectly. Thanks man

**Comment by Johannes-Werbrouck** [comment]
I added a guide for getting Browsershot up and running with laravel Sail on M1 macs here: https://github.com/spatie/browsershot/discussions/711

**Reply by michaelklopf** [reply]
But this only works when configuring Browsershot to work over the network, which would be a thing only for development with sail, but in production it wouldn't be what we need.

**Reply by hexionaut** [reply]
@michaelklopf sail is only made for development purpose from what I know

**Reply by michaelklopf** [reply]
But the calls to Browsershot are there both for dev and prod, which would break on prod, wouldn't they?
